framework/src/timeintegrators/TimeIntegrator.C

```diff
@@ -36,12 +40,36 @@ TimeIntegrator::TimeIntegrator(const InputParameters & parameters)
     _dt(_fe_problem.dt()),
     _dt_old(_fe_problem.dtOld()),
     _Re_time(_nl.getResidualTimeVector()),
-    _Re_non_time(_nl.getResidualNonTimeVector())
+    _Re_non_time(_nl.getResidualNonTimeVector()),
+    _n_nonlinear_iterations(0),
+    _n_linear_iterations(0)
 {
 }
 
 void
 TimeIntegrator::solve()
 {
   _nl.system().solve();
+
+  _n_nonlinear_iterations = getNumNonlinearIterationsLastSolve();
+  _n_linear_iterations = getNumLinearIterationsLastPetscSolve();
+}
+
+unsigned int
+TimeIntegrator::getNumNonlinearIterationsLastSolve() const
+{
+  return _nonlinear_implicit_system->n_nonlinear_iterations();
+}
+
+unsigned int
+TimeIntegrator::getNumLinearIterationsLastPetscSolve() const
```

Comment:
The reason I put `Petsc` in the name is because it only applies if PETSc was used for the nonlinear solver. `ActuallyExplicitEuler`, for example, uses a libMesh solver, and so if someone called `getNumLinearIterationsLastSolve`, they'd get back an incorrect number of linear iterations. Do you still think I should remove it? If so, I can at least put a caution in the doxygen description.

Reply:
I was thinking for the future where this function would give back the number of linear iterations no matter what the solver was...

Could `ActuallyExplicitEuler` override this function and return the right number?

Reply:
This should be possible, but keep in mind that this function doesn't even have `public` access; it is only used internally by `TimeIntegrator` classes. So what would happen here in `ActuallyExplicitEuler` would be this:
```
getNumLinearIterationsLastSolve() {return _n_linear_iterations;}
```
but it wouldn't be used anywhere because `_n_linear_iterations` is already set. Or we'd change
```
_n_linear_iterations = num_its_and_final_tol.first;
```
to
```
_n_linear_iterations_last_solve = num_its_and_final_tol.first;
```
and then do
```
_n_linear_iterations = getNumLinearIterationsLastSolve();
```
where
```
getNumLinearIterationsLastSolve() {return _n_linear_iterations_last_solve;}
```
What would you prefer to do? I'll need to make changes in any case because I need to add requirements, etc.

Reply:
Just for the record, we decided to drop the `Petsc` part from the method name, because we are able to call the solver independent interface in libmesh which is `NonlinearSolver<T>`.